cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -464,11 +467,12 @@ impl MachInstEmit for Inst {
                 x.emit(&[], sink, emit_info, state)
             }
             &Inst::RawData { ref data } => {
-                // emit_island if need, right now data is not very long.
-                let length = data.len() as CodeOffset;
-                if sink.island_needed(length) {
-                    sink.emit_island(length);
-                }
+                // Right now we only put a u32 or u64 in this instruction.
+                // It is not very long, no need to check if need `emit_island`.
+                // If data is very long , this is a bug because RawData is typecial
+                // use to load some data and rely on some positon in the code stream.
+                // and we may exceed `Inst::worst_case_size`.
+                // for more information see https://github.com/bytecodealliance/wasmtime/pull/5612.
                 sink.put_data(&data[..]);
                 // safe to disable code length check.
                 start_off = sink.cur_offset();
```

Comment:
Since we no longer emit the island, we should probably not update this.

Reply:
sure.